CONTRIBUTING.md

```diff
@@ -0,0 +1,254 @@
+with `IMPORTANT:`, otherwise, place it below the other `IMPORTANT` entries. If
+you're not sure if your change merits a news entry, or if it's important,
+please ask.
+
+### Coding Style
+
+In general: Just look at the existing source code!
+
+We try to target experienced JavaScript developers who do not necessarily need
+to have a good understanding of Vimperator's source code, nor necessarily
+understand in-depth concepts of other languages like Lisp or Python. Therefore,
+the coding style should feel natural to any JavaScript developer. Of course,
+this does not mean, you have to avoid all new JavaScript features like list
+comprehension or generators. Use them, when they make sense, but don't use them
+when the resulting code is hard to read.
+
+#### The most important style issues are:
+
+- Use 4 spaces to indent things, no tabs, not 2, nor 8 spaces. If you use Vim,
+  this should be taken care of automatically by the modeline (like the one
+  below).
+
+- No trailing whitespace.
+
+- Use `"` for enclosing strings instead of `'`, unless using `'` avoids
+  escaping of lots of `"`:
+
+  ```javascript
+  alert("foo")
+
+  alert('foo')
+  ```
+
+- Use `//` regexp literals rather than RegExp constructors, unless you're
+  constructing an expression on the fly, or RegExp constructors allow you to
+  escape less `/s` than the additional escaping of special characters required
+  by string quoting:
+
+  ```javascript
+  // Good
+  /application\/xhtml\+xml/
+  RegExp("http://(www\\.)vimperator.org/(.*)/(.*)")
+
+  // Bad
+  RegExp("application/xhtml\\+xml")
+  /http:\/\/(www\.)vimperator.org\/(.*)\/(.*)/
+  ```
+
+- Exactly one space after `if/for/while/catch` etc. and after a comma, but none
+  after a parenthesis or after a function call:
+
+  ```javascript
+  for (pre; condition; post)
+  alert("foo");
+  ```
+
+- Bracing is formatted as follows:
+
+  ```javascript
+  function myFunction () {
+      if (foo)
+          return bar;
+      else {
+          baz = false;
+          return baz;
+      }
+  }
+  var quux = frob("you",
+      {
+          a: 1,
+          b: 42,
+          c: {
+              hoopy: "frood"
+          }
+      });
+  ```
+
+  When in doubt, look for similar code.
+
+- No braces for one-line conditional statements:
+
+  ```javascript
+  if (foo)
+      frob();
+  else
+      unfrob();
+  ```
+
+- Prefer lambda-style functions where suitable:
+
+  ```javascript
+  // Good
+  list.filter(function (elem) elem.good != elem.BAD);
+  list.forEach(function (elem) { window.alert(elem); });
+
+  // Bad
+  list.filter(function (elem) { return elem.good != elem.BAD });
+  list.forEach(function (elem) window.alert(elem));
+  ```
+
+- Anonymous function definitions should be formatted with a space after the
+  keyword `function`:
+
+  ```javascript
+  // Good
+  function () {}
+
+  // Bad
+  function() {}
+  ```
+
+- Prefer the use of `let` over `var` i.e. only use `var` when required.
```

Comment:
All right. What about [developer.mozilla.org - Features not compliant with the ES6 specification][1]?

If it doesn't matter, I was thinking maybe:

> - Adhere to modern JavaScript variable declarations, i.e. prefer `const` 
>   over `let`, and `let` over `var`.
> 
>   For more details, see:
> 
>   - https://hacks.mozilla.org/2015/07/es6-in-depth-let-and-const/
>   - https://developer.mozilla.org/en/New_in_JavaScript_1.7#Block_scope_with_let

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/ECMAScript_6_support_in_Mozilla#Features_not_compliant_with_the_ES6_specification